machine-tests/src/test/java/com/dylibso/chicory/testing/MachinesTest.java

```diff
@@ -279,4 +282,106 @@ public void shouldCallIndirectAotToInterpreter() {
+        //   1: core::panicking::panic_nounwind_fmt::runtime
+        //             at
+        // /rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/panicking.rs:117:22
+        //   2: core::panicking::panic_nounwind_fmt
+        //             at
+        // /rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/intrinsics/mod.rs:3241:9
+        //   3: core::panicking::panic_nounwind
+        //             at
+        // /rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/panicking.rs:218:5
+        //   4: core::slice::raw::from_raw_parts::precondition_check
+        //             at
+        // /rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/ub_checks.rs:68:21
+        //   5: core::slice::raw::from_raw_parts
+        //             at
+        // /rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/ub_checks.rs:75:17
+        //   6: count_vowels
+        //             at ./src/lib.rs:23:26
+        //   7: count_vowels::tests::test_count_vowels
+        //             at ./src/lib.rs:40:9
+
+        // It's not exactly 1-to-1, but it's close enough.
+        assertTrue(
+                exceptionTxt.contains(
+                        "at chicory interpreter 0x00627d:"
+                                + " func.86(library/std/src/panicking.rs:697)"));
+        assertTrue(
+                exceptionTxt.contains(
+                        "at chicory interpreter 0x007e74:"
+                                + " func.118(library/core/src/panicking.rs:117)"));
+        assertTrue(
+                exceptionTxt.contains(
+                        "at chicory interpreter 0x007ec8:"
+                                + " func.119(library/core/src/panicking.rs:218)"));
+        assertTrue(
+                exceptionTxt.contains(
+                        "at chicory interpreter 0x001cf6:"
+                            + " func.30(/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/ub_checks.rs:68)"));
+        assertTrue(
+                exceptionTxt.contains(
+                        "at chicory interpreter 0x003948:"
+                            + " func.54(/rustc/17067e9ac6d7ecb70e50f92c1944e545188d2359/library/core/src/ub_checks.rs:75)"));
+        assertTrue(
+                exceptionTxt.contains("at chicory interpreter 0x000d7f: func.11(src/lib.rs:23)"));
+    }
+
+    @Test
+    public void shouldEmitUnderstandableStackTracesCompiled() throws Exception {
+        WasmModule module = loadModule("compiled/count_vowels.rs.wasm");
+        var instance =
+                Instance.builder(module)
+                        .withDebugParser(DebugParser::parse)
+                        .withMachineFactory(
+                                MachineFactoryCompiler.builder(module)
+                                        .withDebugParser(DebugParser::parse)
+                                        .compile())
```

Comment:
I do wish there was a way we could avoid having to repeat options like `withDebugParser` at the instance and compiler level.

Reply:
feels like something that might go on the Store? 🤔